crates/matrix-sdk/src/client/mod.rs

```diff
@@ -563,13 +589,15 @@ impl Client {
     /// # });
     /// ```
     pub async fn remove_event_handler(&self, handle: EventHandlerHandle) {
+        let key = handle.to_key(None);
         let mut event_handlers = self.inner.event_handlers.write().await;
 
-        if let Some(v) = event_handlers.get_mut(&handle.ev_id) {
+        if let btree_map::Entry::Occupied(mut entry) = event_handlers.entry(key) {
+            let v = entry.get_mut();
             v.retain(|e| e.handle.handler_id != handle.handler_id);
 
             if v.is_empty() {
-                event_handlers.remove(&handle.ev_id);
+                entry.remove();
             }
```

Comment:
Wait, doesn't this mean that the event-handler we've been given by `add_room_event_handler`, which was stored at `handle.to_key(room_id)` wouldn't be removed when calling this now, as we try to look up at `handle.to_key(None)`?

This is confusing, unless there is a separate `remove_room_event_handler` and that is more explicitly called out to be the one used to be remove room-keys ...

But from an API-perspective, given that I am handed an `EventHandlerHandler` by `add_room_event_handler` I would actually not expect to have to call a different function but instead, expect this one to work. How about adding the `OwnedRoomId` to the `EventHandlerHandler`?

Reply:
Yeah you're right, I should have updated this but forgot. Thanks for catching it.